fullstack-cert/react-projects/search-app/index.jsx

```diff
@@ -0,0 +1,96 @@
+const { useState, useMemo, useCallback } = React; // See if we can convert to import statements on Learn
+
+const items = [
+  "Apple",
+  "Banana",
+  "Orange",
+  "Grapes",
+  "Mango",
+  "Pineapple",
+  "Strawberry",
+];
+
+// Global variable to track `selectItem` across renders
+let prevSelectItem = null;
+
+export const SearchApp = () => {
+  const [query, setQuery] = useState("");
+  const [selectedItem, setSelectedItem] = useState(null);
+
+  console.log("Rendering App...");
```

Comment:
is this going to be part of the steps? or was this just for testing purposes?

Reply:
Originally I was thinking of leaving this in, but like you mentioned before, it can get noisy.

Some logging is necessary to see that `useMemo` and `useCallback` are working as expected, but I'll try to clean up the rest.